### PR TITLE
Refactor constructors

### PR DESCRIPTION
**Pull request by omrilit**

### CODE REVIEW

**Comment by omrilit** [comment]
I plan to push this 3 commits as separate commits and not squash them.

**Comment by omrilit** [comment]
Please review along with https://github.com/salto-io/salto/pull/828

File: app/record/lists/list.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ class List extends BaseObject {
 
         const node = {};
 
-        node[type] = {};
+        node[type] = [];
 
         if (attributes) {
             node[type]["$attributes"] = attributes;
```
Comment (ori-moisis) [inline app/record/lists/list.js]: this will be broken now that `node[type]` is a list, no?

Comment (omrilit) [reply app/record/lists/list.js]: I was also concerned from that but it works as expected

**Review by ori-moisis — APPROVED** [review]
I have no idea what is going on here, what is the reason for making this large change or if it will work but I don't think anyone other than you knows anything more so I am going to approve blindly.
I will ask though if we want to diverge so far from the original library? (basically with a change of this size it would probably no longer be feasible to sync from upstream / npm again)
If we are diverging into our own library, maybe we need to release it to npm as well? not sure what is the proper OSS etiquette around this

**Comment by omrilit** [comment]
The reason for this large modification is that the orig library uses lots of specific classes for each type (StringField, BooleanField etc) instead of passing an identifying parameter (this is just a small example). This causes our transformer code to know that when encountering a string field it should use the StringField constructor and it makes the code very messy.

I don't see any added value by sticking to the same folder structure as written originally and I even consider to move all of this logic to our adapter's code.

Regarding the OSS I feel like it's too early to talk about.

**Comment by hadard** [comment]
Is it possible to have a similar code in the adapter package instead of changing `node-suitetalk`? 
As @ori-moisis mentioned this change will probably won't be merged back to `node-suitetalk` so we won't be able to move to newer versions of `node-suitetalk` easily.  In addition, we will need to open source `@salto-io/node-suitetalk`.

if this comment introduces significant work let's keep as is as:
1. we are still not sure what we will do with netsuite
2. `node-suitetalk` has 3 commits and the last one was in Oct 2018

**Pull request merged**